tfhe/src/core_crypto/commons/math/torus/mod.rs

```diff
@@ -97,6 +97,8 @@ pub trait UnsignedTorus:
     + RandomGenerable<Uniform, CustomModulus = Self>
     + Display
     + Debug
+    + Sync
+    + Send
```

Comment:
@tmontaigu opinion on the Sync + Send bound on UnsignedTorus ?

btw @mayeul-zama maybe better on the UnsignedInteger trait, but it's something I really haven't thought about enough

Reply:
Seems ok to me, UnsignedTorus requires UnsignedInteger which require UnsignedNumeric
which finally require Copy, and I don't know if many Copy types are non-Send or non-Sync,
so i don't think adding this will be too restrictive

Reply:
But for big ints this can be an issue right ? like if the UnsignedInteger were marked Sync and Send ?

Reply:
what I mean is UnsignedTorus is related to the mathematical torus we manage and has nothing to do with "programming properties" of the type

Reply:
For 'static' bigint (bigints backed by a `[u64; N]`  which we have in the signed integers PR) this wont be a problem, for 'dynamic' bigint (backed by a `Vec<u64>` which we don't have (but we have all building block)) then UnsignedInteger and Numeric familly of type are not implementatble as they are not `Copy`. Send + Sync should not be a problem
since they are just going to be Vec<u64> without interior mutability thing (Rc, Arc)

If you prefer them to be on the UnsignedInteger trait then its ok as well

I don't have strong opininion on whether these Send + Sync should be UnsignedInteger trait or directly on the function trait bounds that actually need them

Reply:
are we going to need dynamic big ints ?

also if Numeric is marked Copy then maybe the Send + Sync ought to be on the Numeric trait directly ?

Reply:
> are we going to need dynamic big ints ?

It depends, for now we don't, but it could happen

Right now users can't encrypt-decrypt more than U256, with the static big int types they will be able to.
However StaticBigInts might create problems due to their stack size required.

Another example is the scalar div, we added and assert that ensures the scalar type has enough bits of precision compared to the ciphertexts number of bits. If we wanted to do the upcasting ourselves we would require dynamic bigint.

Also if we wanted to have the boolean like enum between real and trivial ciphertext at the `tfhe::integer` level and not at the hlapi level, then we would also need them.

> also if Numeric is marked Copy then maybe the Send + Sync ought to be on the Numeric trait directly ?

seems fair

Reply:
@mayeul-zama if you can add the bound on Numeric rather than UnsignedTorus please :) the existing Send/Sync bounds will stay and disappear over time I guess

Reply:
Done